partiql-ast/src/main/java/org/partiql/ast/AttributeConstraint.java

```diff
@@ -0,0 +1,105 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.expr.Expr;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * TODO docs, equals, hashcode
+ */
+public abstract class AttributeConstraint extends AstNode {
+
+    // NULL & NOT NULL
+    /**
+     * TODO docs, equals, hashcode
+     */
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static class Nullable extends AttributeConstraint {
+
+        public final String name;
```

Comment:
If `name` is common to every variant, then it should go under the abstract class rather than be a field of every variant.

Reply:
Lombok has some limitations on builder annotation with inheritance.

There seems to be work around with the annotation [`SuperBuilder`](https://projectlombok.org/features/experimental/SuperBuilder), but is not really familiar with this...

From their doc: 
> all superclasses also have the @SuperBuilder annotation

Not sure how this would imply with multiple level of hierarchy. 

So if Ok with you, may be we should leave this out of the scope for this CR.